File: geometry_and_trig/triangle_area_heron.py
"""
triangle_area_heron.py
Calculate the area of a triangle using Heron's formula.
Author: Scott
"""

import math

def triangle_area(a, b, c):
    s = (a + b + c) / 2
    return math.sqrt(s * (s - a) * (s - b) * (s - c))

if __name__ == "__main__":
    try:
        a = float(input("Enter side a (default 3): ") or 3)
        b = float(input("Enter side b (default 4): ") or 4)
        c = float(input("Enter side c (default 5): ") or 5)
        area = triangle_area(a, b, c)
        print(f"Triangle area ≈ {area:.4f} square units")
    except ValueError:
        print("Invalid input.")

File: geometry_and_trig/triangle_area_heron.cpp
// triangle_area_heron.cpp - Compute the area of a triangle using Heron's formula.
// Author: Scott

#include <iostream>
#include <cmath>
using namespace std;

double heron_area(double a, double b, double c) {
    double s = (a + b + c) / 2.0;
    return sqrt(s * (s - a) * (s - b) * (s - c));
}

int main() {
    double a = 3, b = 4, c = 5;
    string input;

    cout << "Enter side a (default 3): ";
    getline(cin, input);
    if (!input.empty()) a = stod(input);

    cout << "Enter side b (default 4): ";
    getline(cin, input);
    if (!input.empty()) b = stod(input);

    cout << "Enter side c (default 5): ";
    getline(cin, input);
    if (!input.empty()) c = stod(input);

    double area = heron_area(a, b, c);
    cout << "Area of triangle = " << area << endl;
    return 0;
}
